include/deal.II/lac/la_vector.h

```diff
@@ -0,0 +1,311 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C)  2015 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__la_vector_h
+#define dealii__la_vector_h
+
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/logstream.h>
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/index_set.h>
+#include <deal.II/lac/read_write_vector.h>
+#include <deal.II/lac/vector_space_vector.h>
+
+#include <cstdio>
+#include <iostream>
+#include <cstring>
+#include <vector>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace LinearAlgebra
+{
+  /*! @addtogroup Vectors
+   *@{
+   */
+
+  /**
+   * Numerical vector of data. This class derives from both
+   * ::dealii::LinearAlgebra::ReadWriteVector and
+   * ::dealii::LinearAlgebra::VectorSpaceVector. As opposed to the array of the
+   * C++ standard library, this class implements an element of a vector space
+   * suitable for numerical computations.
+   *
+   * @authod Bruno Turcksin, 2015.
+   */
+  template <typename Number>
+  class Vector : public ReadWriteVector<Number>, public VectorSpaceVector<Number>
+  {
+  public:
+    /**
+     * Constructor. Create a vector of dimension zero.
+     */
+    Vector();
+
+    /**
+     * Copy constructor. Sets the dimension to that of the given vector and
+     * copies all elements.
+     */
+    Vector(const Vector<Number> &V);
+
+    /**
+     * Constructor. Set dimension to @p n and initialize all elements with
+     * zero.
+     *
+     * The constructor is made explicit to avoid accident like this:
+     * <tt>v=0;</tt>. Presumably, the user wants to set every element of the
+     * vector to zero, but instead, what happens is this call:
+     * <tt>v=Vector@<Number@>(0);</tt>, i.e. the vector is replaced by one of
+     * length zero.
+     */
+    explicit Vector(const typename ReadWriteVector<Number>::size_type n);
+
+    /**
+     * Initialize the vector with a given range of values pointed to by the
+     * iterators. This function is there in analogy to the @p std::vector class.
+     */
+    template <typename InputIterator>
+    Vector(const InputIterator first, const InputIterator last);
+
+    /**
+     * Destructor, deallocates memory.
+     */
+    ~Vector();
```

Comment:
Isn't the destructor of the `VectorSpaceVector` class virtual? If so, you should mark this as virtual as well.